### PR TITLE
Add metrics about disk conditions

### PR DESCRIPTION
**Pull request by petertakacs**
New metric is created called longhorn_disk_status. Works similary as longhorn_node_status, it exposes information about disk by adding conditions as metric labels.

With this metric one can setup alerts in prometheus in case disk is not schedulable

resolves https://github.com/longhorn/longhorn/issues/6858

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM

**Comment by innobead** [comment]
@c3y1huang Please help review this and drive https://github.com/longhorn/longhorn/issues/6858 to go through the QA process. Thanks.

**Comment by innobead** [comment]
@mergify backport v1.4.x v1.5.x

File: metrics_collector/disk_collector.go [hunk 1]
```diff
@@ -1,8 +1,10 @@
 package metricscollector
 
 import (
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/sirupsen/logrus"
+	"strings"
```
Comment (c3y1huang) [inline metrics_collector/disk_collector.go:7]: nit: consider [organize package imports](https://github.com/longhorn/longhorn/wiki/coding-convention#organize-package-imports).

**Review by c3y1huang — APPROVED** [review]
lgtm

**Comment by innobead** [comment]
@mergify backport v1.5.x v1.4.x

File: metrics_collector/disk_collector.go (hunk 1 above)
Comment (innobead) [reply metrics_collector/disk_collector.go:7]: @c3y1huang let's improve this later.

Comment (c3y1huang) [reply metrics_collector/disk_collector.go:7]: https://github.com/longhorn/longhorn-manager/pull/2230

**Pull request merged**